### ISSUE TITLE
[Bug] disallow-capabilities-strict policy does not conform to Pod Security Admission implementation or PSS spec

### ISSUE THREAD

**Issue by joebowbeer** [kyverno/policies#450]
### Kyverno Version

1.8.1

### Kubernetes Version

1.25.x

### Kubernetes Platform

KinD

### Description

The kyverno PSS/restricted policy will allow any mixed-case spelling of 'ALL' in securityContext.capabilities.drop['ALL']

For example, any of these are accepted: `all` | `ALL` | `aLL` | `All` | and so on

However, the PSS/restricted [spec](https://kubernetes.io/docs/concepts/security/pod-security-standards/) explicitly requires 'ALL' and the Pod Security Admission controller [implementation](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/pod-security-admission/policy/check_capabilities_restricted.go) requires 'ALL' as well.

I view this as a regression introduced by #421

In practice, cri-o and containerd appear to do case-insensitive matching for the `ALL` keyword, but this is not guaranteed to work for other runtimes.

Until a stricter CRI capabilities spec is championed in SIG-Node, it's preferable and arguably more correct for the PSS policies to stick to the canonical `ALL` capitalization. That way they will be consistent with each other as well as their own spec. It is very unlikely that the PSA implementation will change without a corresponding change to the CRI capabilities spec.

### Steps to reproduce

Create v1.25 kind cluster

```shell
kind create cluster
```

Create `test-drop-all` namespace

```shell
kubectl create ns test-drop-all
```

Create nginx pod in the test-drop-all namespace

Note: securityContext.capabilities.drop['all']

```shell
kubectl apply -n test-drop-all -f - <<EOF
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
    - image: nginxinc/nginx-unprivileged
      name: nginx
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop: ['all']
        runAsNonRoot: true
        seccompProfile:
          type: RuntimeDefault
EOF
```

Observe that the built-in (PSA) pod-security enforcement does not accept this:

```shell
kubectl label --dry-run=server --overwrite ns test-drop-all pod-security.kubernetes.io/enforce=restricted
```

> Warning: existing pods in namespace "test-drop-all" violate the new PodSecurity enforce level "restricted:latest"
> Warning: nginx: unrestricted capabilities

But it passes Kyverno's pod-security policies:

```shell
kyverno apply <(kustomize build https://github.com/kyverno/policies/pod-security) -n test-drop-all -c
```

> Applying 19 policy rules to 1 resource...
> pass: 19, fail: 0, warn: 0, error: 0, skip: 38

### Expected behavior

Kyverno's PSS/restricted enforcement should match that of PSA.

Esp. if I'm using kyverno CLI to pre-validate manifests for my cluster, I expect kyverno to be at least as restrictive as the PSA policies in the cluster.

### Slack discussion

https://kubernetes.slack.com/archives/C032MM2CH7X/p1667224893686709

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by joebowbeer** [comment]
@chipzoller Based on feedback I received from a PSA implementer, I recommend reverting the change to kyverno/policies/pod-security in #421 and the similar change to kyverno/kyverno/charts in https://github.com/kyverno/kyverno/pull/4954

I would, though, retain the case-insensitive handling in the best practices policy - as an example of how to express this leniently.

Regarding kyverno's pod-security policies, there is no requirement for case insensitivity in the runtime specs (unfortunately?), but case-**_sensitivity_** is specified in the PSS/restricted spec and implemented in PSA.

See description for full rationale.

I can create PRs if this is agreeable.

**Comment by chipzoller** [comment]
What is harm in leaving them as is? It has no negative functional impact from what I can tell.

**Comment by joebowbeer** [comment]
@chipzoller You're asking what is wrong with not implementing the spec? 😄 

FWIW, The PSA maintainer I spoke to would prefer that all the pod-security implementations match each other and the spec.

I think there are several possible "harms" that can come from this:

1. The most likely one to affect me is that I'm using kyverno CLI to pre-validate manifests for my cluster. If kyverno accepts a manifest that PSA rejects (see example in description) then I get a failure at runtime instead of during development.

2. The potential harm is that some CRI-compliant runtime **_will be_** case-sensitive. Or containerd will tighten its implementation... Then, by validating my manifest, kyverno may overlook a potential runtime vulnerability - or failure.

**Comment by chipzoller** [comment]
I don't see a problem with walking these back. I think it would be helpful for there to be some corresponding documentation on the Kubernetes PSA side to which we could point users for an "official" statement lest it seems like it's a Kyverno gap/oversight. If you'd like to propose at least the Kyverno side in PRs, that would be grand.

**Comment by chipzoller** [comment]
This is being addressed in PR #485
